src/app/screens/Register/index.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import i18next from 'i18next';
+import { useForm } from 'react-hook-form';
+
+import Input from '../../components/Input';
+import Button from '../../components/Button';
+import imageWolox from '../../assets/wolox.svg';
+import { emailRegex } from '../../../utils/inputValidations';
+
+import styles from './styles.module.scss';
+
+interface DataForm {
+  email: string;
+  firstName: string;
+  lastName: string;
+  password: string;
+  passwordConfirmation: string;
+  locale: string;
+}
+
+function Register() {
+  const { register, handleSubmit, errors, formState, watch } = useForm<DataForm>({
+    mode: 'all'
+  });
+
+  const [language, setLanguage] = useState(i18next.language);
+
+  const submitForm = (data: DataForm) => {
+    data.locale = language;
+    const user = {
+      user: data
+    };
+    console.log(user);
+  };
+
+  const validatePassword = (passwordConfirmation: string) =>
+    passwordConfirmation === watch('password') ? true : 'Register:passwordNotMatch';
+
+  const changeLang = () => {
+    const translationTo = i18next.language === 'es' ? 'en' : 'es';
+
+    i18next.changeLanguage(translationTo, () => {
+      setLanguage(i18next.language);
+    });
+  };
```

Comment:
nice but unnecessary.
also the `changeLanguage` listener could be place on config, and the state is useless because it could be asked to i18next anytime fyi for the next time you use this